File: algorithms/strings/get_digits/get_digits.js
function get_digits(myString) {
  var len = myString.length;
  var newString = "";
  var newInteger;
  for (let i = 0; i < len; i++) {
    if (myString[i].match(/\d/)) {
      newString += myString[i];
    }
  }
  newInteger = parseInt(newString);
  return newInteger;
}

var myString = "0s1a3y5w7h9a2t4?6!8?0"

console.log(get_digits(myString));

File: algorithms/strings/get_digits/get_digits.py
import re

def get_digits(myString):
    ln = len(myString)
    newString = ""
    pattern = re.compile("\d")

    for i in range(ln):
        if (pattern.match(myString[i])):
            newString += myString[i]

    return int(newString)

myString = "0s1a3y5w7h9a2t4?6!8?0"
print(get_digits(myString))
